core/app/models/spree/product.rb

```diff
@@ -323,7 +323,7 @@ def taxons_for_store(store)
     end
 
     def full_in_stock?
-      variants.in_stock_or_backorderable.exists?
+      variants_including_master.in_stock_or_backorderable.exists?
```

Comment:
@sebastian-palma I think we need to cover this in a different way - the usual setup is that if you have variants other than the master (hate that name!) then you should only check these non-master variants because after adding a first-non master variant, the master is set out of stock

eg. `Product#purchasable` method - 

https://github.com/spree/spree/blob/8809bed09a7a6cd5ed61e2534a62f2e98219a5f7/core/app/models/spree/product.rb#L146